ui/apps/platform/src/services/ClustersService.ts

```diff
@@ -97,8 +97,14 @@ export function fetchClusterWithRetentionInformation(id: string): Promise<Cluste
     });
 }
 
+export enum Upgradability {
+    ALLOWED = 'ALLOWED',
+    NOT_ALLOWED = 'NOT_ALLOWED',
+}
```

Comment:
Although not a blocker to merge, string union type is more fluent TypeScript than enum type.

Historical note: main reason for enum type was at beginning of migration from JavaScript to TypeScript, calling code might have used an enum-like idiom using objects and properties. Because UI is making steady progress to TypeScript `'ALLOWED'` is simpler and more idiomatic.

Reply:
I will comment here: is it possible to delete `?` from `enableAutoUpgrade` property below?

Reply:
Done